### PR TITLE
feat(ST-23): add landing page and role choosing modal

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change
- Added a HomePage module
- Added two components for Welcome message and for role choosing modal
- Added global background color at _app.page.tsx

## Associated Ticket Link(s)

- https://trello.com/c/PrtZQ6m5

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/c6891f4e-5253-48f0-a15f-405208b717c9

Tablet View:
Screen Shot 2024-08-21 at 10 52 44 AM
Screen Shot 2024-08-21 at 10 52 33 AM

Mobile View:
Screen Shot 2024-08-21 at 10 49 33 AM

Screen Shot 2024-08-21 at 10 51 00 AM

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "noImplicitOverride": true,
     "noImplicitReturns": true,
     "noImplicitThis": true,
-    "noPropertyAccessFromIndexSignature": true,
+    "noPropertyAccessFromIndexSignature": false,
```
Comment (ishmam-iqbal-sazim) [inline tsconfig.json:26]: Let's not touch the typescript config file we get from the template.
Best to follow the patterns it enforces.

File: modules/HomePage/components/ChooseRole/ChooseRole.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Box, Flex, Modal, Text, Title } from "@mantine/core";
+import { LuGraduationCap } from "react-icons/lu";
+import { TbBooks } from "react-icons/tb";
+
+import classes from "./ChooseRole.module.css";
+
+interface IChooseRoleProps {
+  opened: boolean;
+  close: () => void;
+}
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/ChooseRole/ChooseRole.tsx:10]: Should be in a separate types file.

File: modules/HomePage/components/Greetings/Greetings.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+import { Box, Button, Flex, Text, Title } from "@mantine/core";
+
+interface IGreetingsProps {
+  open: () => void;
+}
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/Greetings/Greetings.tsx:7]: Same comment as above

File: modules/HomePage/containers/HomeContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useDisclosure } from "@mantine/hooks";
+
+import ChooseRole from "../components/ChooseRole/ChooseRole";
+import Greetings from "../components/Greetings/Greetings";
+
+const HomeContainer = () => {
+  const [opened, { open, close }] = useDisclosure(false);
+  return (
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/containers/HomeContainer.tsx:8]: [NIT] Should be a line break above

File: pages/_app.page.tsx [hunk 5]
```diff
@@ -21,7 +21,19 @@ export default function App(props: TCustomAppProps) {
...
+      <MantineProvider
+        withGlobalStyles
+        withNormalizeCSS
+        theme={{
+          ...appTheme,
+          globalStyles: (theme) => ({
+            body: {
+              backgroundColor: "#00224D",
+              color: theme.white,
+            },
```
Comment (ishmam-iqbal-sazim) [inline pages/_app.page.tsx:33]: Can put this in the appTheme configuration file rather than overriding object values here.

**Review by ishmam-iqbal-sazim — APPROVED** [review]

**Comment by ragib-hossain-sazim** [comment]
Please add screenshots for Mobile and Tablet view as well.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**